File: Abstract Factory Pattern/1. Minimal/Java/Main.java
public class Main {
    public static void main(String[] args) {
        System.out.println("Abstract Factory Pattern in Java");
        System.out.println("--------------------------------");
        System.out.println();
  
        createProducts(new ConcreteFactory1());
        createProducts(new ConcreteFactory2());
    }

    private static void createProducts(AbstractFactory concreteFactory) {
        AbstractProductA productA = concreteFactory.createProductA();
        AbstractProductB productB = concreteFactory.createProductB();

        productA.build();
        productB.build();
    }
}

File: Abstract Factory Pattern/1. Minimal/Python/main.py
from concrete_factory_1 import ConcreteFactory1
from concrete_factory_2 import ConcreteFactory2

def main():
    print("Abstract Factory Pattern in Python")
    print("----------------------------------")
    print()

    createProducts(ConcreteFactory1())
    createProducts(ConcreteFactory2())

def createProducts(concreteFactory):
    productA = concreteFactory.createProductA()
    productB = concreteFactory.createProductB()

    productA.build()
    productB.build()

if __name__ == "__main__":
    main()
